src/redux/reducer.js

```diff
@@ -61,17 +65,56 @@ function loadApi(state = initialState.get('loadApi'), action) {
   }
 }
 
+function setNextCodePosition(state, dir) {
+  let nextCodePosition = state.get('codePosition');
+  const history = state.get('history');
+  nextCodePosition += dir;
+  while (nextCodePosition > 0 &&
+         nextCodePosition < history.size &&
+         history.getIn([nextCodePosition, 'code']) === '') {
+    nextCodePosition += dir;
+  }
+  const nextCode = history.getIn([nextCodePosition, 'code']);
+  if (nextCodePosition < 0 ||
```

Comment:
this lengthy conditional is needed otherwise it will try to access prev code that doensn't exist, and the while statement accounts for the chance that in the future blank code may be recorded in the history